pymbolic/mapper/__init__.py

```diff
@@ -366,40 +366,60 @@ def map_function_symbol(self, expr, *args, **kwargs):
         return expr
 
     def map_call(self, expr, *args, **kwargs):
-        return type(expr)(
-                self.rec(expr.function, *args, **kwargs),
-                tuple(self.rec(child, *args, **kwargs)
-                    for child in expr.parameters))
+        function = self.rec(expr.function, *args, **kwargs)
+        parameters = tuple(self.rec(child, *args, **kwargs)
+                    for child in expr.parameters)
+        if (id(function) == id(expr.function)
+            and all(id(child) == id(orig_child) for child, orig_child in
```

Comment:
This misses that the two iterables could be of different lengths.

Reply:
By construction the two iterables have the same length, right? `parameters` is created from `expr.parameters`.

Reply:
Ah, fair point. :facepalm: Add a comment?